frontend/src/app/map-view/map/map.component.ts

```diff
@@ -200,16 +195,43 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         this.map, this.dispatchSelectionUpdate, this.zoomToExtent,
         this.onDrawEnd, this.onDrawInvalid, this.onDownloadClick, this.onSplitClick, this.onMergeClick,
         this.scenarioLayer, this.translateService, this.geoJson); // Will add itself to the map
-    this.map.addLayer(this.areaLayer);
 
+    this.areaHighlightLayer = new AreaHighlightLayer(this.map, this.geoJson);
+
+    this.areaLayer.setBoundaries(boundaries);
+
+    this.areaSubscription = this.store.select(AreaSelectors.selectAreaFeatures)
```

Comment:
I spent a good while trying to make this not be a subscription before I gave up, essentially. I think it obviously shouldn't be, since we'd only need to "map" the selectable areas once at startup.
I stopped myself from trying a `mergeAll` operation inline on two different selectors (instead of the combined `selectAreaFeatures` used here. Even if that would work (could it?) I feel that 'code smell' would be too overwhelming.
This should all be remedied by fixing the `AreaLayer` component implementation, instead of introducing more or less bulky workarounds. See the PR description about that.